diff --git a/common/keyframe.cpp b/common/keyframe.cpp
index 67b97b0..6aca17e 100644
--- a/common/keyframe.cpp
+++ b/common/keyframe.cpp
@@ -63,8 +63,8 @@ typedef struct
 
 #pragma pack(pop)
 
-#define INITIAL_BIG_SHAPE_BUFFER_SIZE 12000 * 1024
-#define THEATER_BIG_SHAPE_BUFFER_SIZE 2000 * 1024
+#define INITIAL_BIG_SHAPE_BUFFER_SIZE 3000 * 1024
+#define THEATER_BIG_SHAPE_BUFFER_SIZE 1000 * 1024
 #define UNCOMPRESS_MAGIC_NUMBER       56789
 
 static unsigned short CurrentUncompressMagicNum = UNCOMPRESS_MAGIC_NUMBER;
